framework/sandstone.cpp

```diff
@@ -2685,6 +2723,112 @@ static void warn_deprecated_opt(const char *opt)
             program_invocation_name, opt);
 }
 
+
+static bool system_is_idle(bool skip_wait, float idle)
+{
+    if(skip_wait == true)
+        return true;
+
+    #ifdef __linux__
+
+    FILE *loadavg;
+    float load_1, load_5, load_15;
+    int ret;
+
+    // Look at loadavg for hints whether the system is reasonably idle or not
+    // any error and we assume it's busy/under load for simplicity
+
+    loadavg = fopen("/proc/loadavg", "r");
+    if (!loadavg)
+        return false;
+
+    ret = fscanf(loadavg, "%f %f %f", &load_1, &load_5, &load_15);
+    fclose(loadavg);
+
+    if (ret != 3)
+       return false;
+
+    if (load_5 <= idle)
+       return true;
+    #else //__linux__
+    return true;//TO DO implement idle trigger for Windows
+    #endif
+
+    return false;
+}
+
+static void background_scan_init(void)
+{
+    uint64_t t_now = get_current_time();
+
+    // init timestamps to 1day old - this quickstarts testing on first run
+    if ((sApp->background_scan.init_timestamp == true) && (sApp->service_background_scan == true)) {
+        for (auto i = 0; i < sApp->background_scan.number_of_timestamps; i++) {
+            uint64_t timestamp_init_val = 
+                (t_now - sApp->background_scan.day_delay_between_tests.count());
+            sApp->background_scan.timestamp.push_front(timestamp_init_val);
+        }
+    }
+}
+
+static void background_scan_update_load_threshold(void)
+{
+    uint64_t t_now = get_current_time();
+
+    // scale our idle threshold value from 0.2 base, to 0.8 after 12h
+    // every hour adds 0.05 to the threshold value
+    sApp->background_scan.load_idle_threshold = 
+        sApp->background_scan.load_idle_threshold_init +
+        (((t_now - sApp->background_scan.timestamp.back()) / sApp->background_scan.number_seconds_in_hour) * 
+        sApp->background_scan.load_idle_threshold_inc_val);
+
+    // prevent the idle threshold form rising above 0.8
+    if(sApp->background_scan.load_idle_threshold > sApp->background_scan.load_idle_threshold_max)
+        sApp->background_scan.load_idle_threshold = sApp->background_scan.load_idle_threshold_max;
+}
+
+static void background_scan_daemon(void) 
```

Comment:
Misleading function name. Can you rename to something like `background_scan_wait()`?